src/group_theory/group_ring.lean

```diff
@@ -0,0 +1,227 @@
+/-
+Copyright (c) 2021 Amelia Livingston. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Amelia Livingston
+-/
+
+import algebra.monoid_algebra.basic
+import tactic.basic
+
+/-!
+# Group rings
+
+This file defines the group ring `k[G]` of a group `G` with coefficients in a ring `k`. A
+`G`-module with a compatible `k`-module structure is an `R[G]`-module.
+
+We develop an API allowing us to show `k[Gⁿ]` is a free `k[G]`-module for `n ≥ 1`. This
+will be used to construct a projective resolution of the trivial `k[G]`-module `k`.
+
+## Implementation notes
+
+Although `group_ring k G` is just `monoid_algebra k G`, we make the definition `group_ring`
+so we can separate material relevant to group cohomology into the namespace `group_ring`.
+We make it an abbreviation because we want the instances on `monoid_algebra` to apply to
+`group_ring` without code duplication.
```

Comment:
Could this be expanded? To me the file as it is seems a bit strange as everything in this file is about monoids, but is called a group ring.